### ISSUE TITLE
Wondering why `getUser` is async

### ISSUE THREAD

**Issue by kevinbai0** [auth0/auth0-spa-js#684]
I'm curious as to why `getUser` in `Auth0Client` is async and (resulting it to be wrapped in a promise) since it doesn't seem to be operating under any asynchronous code. It would be super nice to have `getUser` to not be wrapper in a promise which would it easier to use.

It'd then let `isAuthenticated` to also be synchronous and even a `getter` which would be a really good nice-to-have.

https://github.com/auth0/auth0-spa-js/blob/171dfce856bff54120668af977ee9da90d3c739b/src/Auth0Client.ts#L422

**Comment by frederikprijck** [comment]
Thanks for reporting. We are aware of this, and this is something we would like to address as part of a next major release, as changing this introduces a breaking change.

However, as we do not want to create a breaking change only for these things, we are planning to batch some breaking changes together in the future.

**Comment by kevinbai0** [comment]
Sounds good, thanks!

**Comment by mattjennings** [comment]
Any idea when this will be addressed? This has been a bit of a pain to work around. 

In an earlier version I was able to get around it like this:

```js
const cache = auth0Client.cache.get({
  client_id: auth0Client.options.client_id,
  audience,
  scope
})

```

which is what getUser did internally at the time. However, now you have to pass it a `CacheKey` instance, which is not an exported class that I can use.

https://github.com/auth0/auth0-spa-js/blob/171dfce856bff54120668af977ee9da90d3c739b/src/Auth0Client.ts#L452-L458

This was all pretty hacky anyway, but it's frustrating that this is only needed because getUser is incorrectly async.

**Comment by frederikprijck** [comment]
Hi @mattjennings,

We are currently working on a PR to make the cache extensible, see https://github.com/auth0/auth0-spa-js/pull/743
This is for the purpose of supporting Ionic with this SDK.
As a result of that, the cache will need to respect an async API, so I believe `getUser` will need to keep using that same async API as well.

What's the reason you need to resort to the above solution? Our cache being async shouldn't require you to access it directly yourself, even more so that it won't work anymore once the above PR is merged because even then our `cache.get()` would be asynchronouss.

**Comment by mattjennings** [comment]
Ah I see. Admittedly my use-case was not very typical, but it was helpful to have the user session synchronously on page load without needing to await getUser and handle the brief loading state. It's probably better to handle that loading state more wisely, but seeing that the underlying cache was synchronous is why I went for the workaround. If it'll be async by nature soon then it's all a moot point!
